node/src/main/scala/org/bitcoins/node/Node.scala

```diff
@@ -159,30 +183,42 @@ trait Node extends NodeApi with ChainQueryApi with P2PLogger {
     val chainApiF = startConfsF.flatMap(_ => chainApiFromDb())
 
     val startNodeF = {
-      val isInitializedFs = peerMsgSenders.indices.map { idx =>
-        peerMsgSenders(idx).connect()
-        val isInitializedF = for {
-          _ <- AsyncUtil.retryUntilSatisfiedF(() => isInitialized(idx),
-                                              maxTries = 200,
-                                              interval = 250.millis)
-        } yield ()
-        isInitializedF.failed.foreach(err =>
-          logger.error(
-            s"Failed to connect with peer=${peers(idx)} with err=$err"))
-        isInitializedF.map { _ =>
-          nodeAppConfig.nodeType match {
-            case NodeType.NeutrinoNode => {
-              if (peerServices(peers(idx)).nodeCompactFilters) {
-                logger.info(s"Our peer=${peers(idx)} has been initialized")
-              } else {
-                logger.info(
-                  s"Our peer=${peers(idx)} does not support compact filters. Disconnecting.")
-                peerMsgSenders(idx).disconnect()
+      logger.info(peers)
+      val isInitializedFs = peers.map { peer =>
+        peerData(peer).peerMessageSender.connect()
```

Comment:
Can we move this lambda implementation into a helper method with a nice name and possibly scaladoc? I.e. 

```scala
private def initializePeer(peer): Future[Unit] = { ... }

private def createInDbIfBlockFilterPeer: Future[Option[PeerDb]] = { ... } 

```

it always makes code so much more readable with nicely named methods :-)